FilePersistence/src/version_control/Commit.cpp

```diff
@@ -107,7 +107,7 @@ bool Commit::isValidMatch(const char* content, qint64 size, const char* indexOfI
 								  bool findChildrenByParentId) const
 {
 	start = indexOfId-content;
-	end = start;
+	end = start+1;
```

Comment:
indexOfId points at '{' in the id "{ id }" and in the later part we make a check 
`// String is of the form {id} {*.}
if (findChildrenByParentId && content[end] == '{') return false;`

then it was always returning false since end was also pointing to '{'
So, to correct that I kept end = start + 1